astropy/coordinates/builtin_frames/utils.py

```diff
@@ -34,29 +36,42 @@
 _DEFAULT_PM = (0.035, 0.29)*u.arcsec
 
 
+def matrix_product(*args):
+    """Matrix multiply all arguments together.
+
+    Arguments should have dimension 2 or larger, with larger dimensional
+    objects being interpreted as stacks of matrices.
```

Comment:
Perhaps take a prompt from the ```numpy.matmul``` documentation and change this to "stacks of matrices residing in the last two indexes"?